### PR TITLE
MEGA65: Add nybl16 colour mode

### PR DESCRIPTION
**Pull request by hernandp**
Nybl16-color plus your own fixes.

### CODE REVIEW

**Comment by lgblgblgb** [comment]
Hmm, something went wrong here :( It seems some TEXTUTRE_ macros wanted to be renamed SCREEN_ ... again :-O

File: targets/mega65/vic4.c [hunk 1]
```diff
@@ -243,18 +243,20 @@ static void vic4_update_sideborder_dimensions ( void )
 	if (REG_CSEL) {	// 40-columns?
 		border_x_left = FRAME_H_FRONT + SINGLE_SIDE_BORDER;
 		if (!REG_H640)
-			border_x_right = FRAME_H_FRONT + TEXTURE_WIDTH - SINGLE_SIDE_BORDER - 1;
```
Comment (lgblgblgb) [inline targets/mega65/vic4.c:246]: Here for example, your PR want to rename TEXTURE_WIDTH back to SCREEN_WIDTH, though in theory you should have for now the state where SCREEN_WIDTH (and HEIGHT) is renamed to TEXTURE...

**Comment by lgblgblgb** [comment]
https://github.com/lgblgblgb/xemu/pull/239/files#r613295124

File: targets/mega65/vic4.c (hunk 1 above)
Comment (hernandp) [reply targets/mega65/vic4.c:246]: Uhmmm... i dont even remember touching that...

Comment (lgblgblgb) [reply targets/mega65/vic4.c:246]: I'm a bit clueless, to be honest :( OK, maybe it was not a good idea from me to merge my changes first to you, but it still does not explain, how things changed back to SCREEN_ ... when I have no such string in the whole mega65 subtree any more ...

Comment (hernandp) [reply targets/mega65/vic4.c:246]: Something got mixed, in my uncomitted vic4.h there were SCREEN_WIDTH and TEXTURE_WIDTH both present. Now I have the synced vic4.h with only TEXTURE_WIDTH present, changed the only 3 occurrences in vic4.c and all it's going well.  Go figure what happened.

File: targets/mega65/vic4.c [hunk 2]
```diff
@@ -1065,6 +1063,22 @@ static void vic4_render_fullcolor_char_row ( const Uint8* char_row, int glyph_wi
 }
```
Comment (lgblgblgb) [inline targets/mega65/vic4.c:1065]: Sorry for the question again, just make it clear :) So the part BELOW this comment looks like what I expected. What about the changes above? Are those really intended? I can't see it's connected to the implementation of the nybl mode video.

Comment (hernandp) [reply targets/mega65/vic4.c:1065]: Looking at the file history, that comments were suited for better debugging the NTSC/PAL mode change.

Comment (lgblgblgb) [reply targets/mega65/vic4.c:1065]: Yes but the !in_hypervisor again changed position what I worry for, but if you say it's intended that way, of course I believe you ;)

Comment (hernandp) [reply targets/mega65/vic4.c:1065]: I think it's the correct position -- may be the performance regression was due to this? e.g:  recalculating modes on every frame access.   The regressed output should have a lot of VIC4: border top=xxx etc.

**Comment by lgblgblgb** [comment]
OK, I've merged hopefully it's fine ;) I'll check the performance now etc, what you guessed about the problem. Thanks.

**Pull request merged**